packages/core-components/src/components/multiselect-dropdown/multiselect-dropdown.tsx

```diff
@@ -50,17 +50,21 @@ export class B2bMultiSelectDropdown {
   @State() hasOptionList = this.optionsList.length > 0;
 
   componentWillLoad() {
-    this.currentSelectedValues = this.selectedValues.filter(value =>
-      this.optionsList.includes(value),
+    this.parseProps();
+    this.currentSelectedValues = (this.selectedValues as string[]).filter(
+      value => this.optionsList.includes(value),
     );
   }
 
   /** Needed to trigger a re-render for async data */
   @Watch('optionsList')
-  watchPropHandler(newList: string[]) {
-    this.hasOptionList = newList.length > 0;
+  watchPropHandler(newList: string | string[]) {
+    if (typeof newList === 'string') {
```

Comment:
Is this required as the options are already parsed during component load? The new list will always be an array as it was parsed during initial render